### ISSUE TITLE
Analog of hana::template_ for lifting non-type templates

### ISSUE THREAD

**Issue by invexed** [boostorg/hana#455]
Is an analog of ```hana::template_``` for lifting non-type templates within the scope of Boost.Hana? There is a naive implementation in my answer here: [https://stackoverflow.com/a/57284483/11817970](url).

**Comment by ldionne** [comment]
Yes, it is definitely in scope. It's simply that we don't currently use C++17 features, however we could/should.

**Comment by invexed** [comment]
I see. A while ago there was some talk about a Hana 2.0, with breaking changes. Is that still in the pipeline? If so, I imagine that would be a good time to incorporate the above.
